src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -205,18 +364,30 @@ class ShenandoahFreeSet : public CHeapObj<mtGC> {
   // ShenandoahOldCompactionReserve.  In a future planned enhancement, the reserve for old_collector set when the
   // evacuation reserves are unknown, is based in part on anticipated promotion as determined by analysis of live data
   // found during the previous GC pass which is one less than the current tenure age.
-  void rebuild(size_t young_cset_regions, size_t old_cset_regions, bool have_evacuation_reserves = false);
-
-  void move_collector_sets_to_mutator(size_t cset_regions);
-
-  void add_old_collector_free_region(ShenandoahHeapRegion* region);
+  void finish_rebuild(size_t young_cset_regions, size_t old_cset_regions, size_t num_old_regions,
+                      bool have_evacuation_reserves = false);
+
+  // When a region is promoted in place, we add the region's available memory if it is greater than plab_min_size()
+  // into the old collector partition by invoking this method.
+  void add_promoted_in_place_region_to_old_collector(ShenandoahHeapRegion* region);
+
+  // Move up to cset_regions number of regions from being available to the collector to being available to the mutator.
+  //
+  // Typical usage: At the end of evacuation, when the collector no longer needs the regions that had been reserved
+  // for evacuation, invoke this to make regions available for mutator allocations.
+  //
+  // Note that we plan to replenish the Collector reserve at the end of update refs, at which time all
+  // of the regions recycled from the collection set will be available.  In the very unlikely event that there
+  // are fewer regions in the collection set than remain in the collector's free set, we limit the transfer in order
+  // to assure that the replenished Collector reserves can be sufficiently large.
```

Comment:
Apropos my previous comment, this part would seem to belong in the caller(s?) of the method.

Reply:
I'm moving a variation of this comment to the callers.  Thanks.